components/eamxx/src/physics/mam/impl/mam4_amicphys.cpp

```diff
@@ -1,1202 +1,1486 @@
-        // if config.gaexch_h2so4_uptake_optaa == 2, then
-        //    if qgas increases from pre-gaschem to post-cldchem,
-        //       start from the pre-gaschem mix-ratio and add in the production
-        //       during the integration
-        //    if it decreases,
-        //       start from post-cldchem mix-ratio
-        // *** currently just do this for h2so4 and nh3
+  // qgas1, qgas3 [kmol/kmol]
+  // qgas_cur     [kmol/kmol]
+
+  // qnum3    [#/kmol]
+  // qnum_cur [#/kmol]
+
+  // qaer2, qaer3 [kmol/kmol]
+  // qaer_cur[kmol/kmol]
+
+  // qnumcw3[#/kmol]
+  // qnumcw_cur  [#/kmol]
+
+  // qaercw2, qaercw3 [kmol/kmol]
+  // qaercw_cur  [kmol/kmol]
+
+  // qwtr3      [kmol/kmol]
+  // qwtr_cur   [kmol/kmol]
+
+  // qXXX_delaa are TMR changes (increments, not tendencies) of different
+  // microphysics processes. These are diagnostics sent to history output;
+  // they do not directly affect time integration.
+
+  // qgas_delaa   [kmol/kmol]
+  // qnum_delaa   [   #/kmol]
+  // qaer_delaa   [kmol/kmol]
+  // qnumcw_delaa [   #/kmol]
+  // qaercw_delaa [kmol/kmol]
+
+  // type ( misc_vars_aa_type ), intent(inout) :: misc_vars_aa_sub
+
+  //---------------------------------------------------------------------------------------
+  // Calculate air molar density [kmol/m3] to be passed on to individual
+  // parameterizations
+  //---------------------------------------------------------------------------------------
+  // BAD CONSTANT
+  // Universal gas constant (J/K/kmol)
+  constexpr Real r_universal = 8314.46759100000;
+  const Real aircon          = pmid / (r_universal * temp);
+
+  //----------------------------------------------------------
+  // Initializ mixing ratios with the before-amicphys values
+  //----------------------------------------------------------
+
+  copy_1d_array(max_gas(), qgas3,  // in
+                qgas_cur);         // out
+
+  constexpr int nspecies = num_aerosol_ids;
+  constexpr int nmodes   = num_modes;
+
+  copy_2d_array(nspecies, nmodes, &qaer3[0][0],  // in
+                &qaer_cur[0][0]);                // out
+
+  copy_1d_array(nmodes, qnum3,  // in
+                qnum_cur);      // out
+
+  copy_1d_array(nmodes, qwtr3,  // in
+                qwtr_cur);      // out
+
+  if(iscldy_subarea) {
+    copy_1d_array(nmodes, qnumcw3,                   // in
+                  qnumcw_cur);                       // out
+    copy_2d_array(nspecies, nmodes, &qaercw3[0][0],  // in
+                  &qaercw_cur[0][0]);                // out
+  }                                                  // iscldy_subarea
+
+  //---------------------------------------------------------------------
+  // Diagnose net production rate of H2SO4 gas production
+  // cause by other processes (e.g., gas chemistry and cloud chemistry)
+  //---------------------------------------------------------------------
+  Real qgas_netprod_otrproc[max_gas()] = {0};
+  assign_1d_array(max_gas(), 0.0,         // in
```

Comment:
This is pointless. You are already setting all entries to 0 in the line above.

Reply:
@singhbalwinder, you mentioned that you had to assign zero using a for loop because this type of operation, Real qgas_netprod_otrproc[max_gas()] = {0}, was only assigning the first entry. I have observed this behavior as well. Are we missing something?

Reply:
I am also confused about it. I chatted with @bartgol about it today, and he mentioned that compilers guarantee that all elements will be zero. In my experience, I sometimes found that not to be true. That is why, I am doing these explicit assignments. I am not sure what we might be missing.